eclass/toolchain.eclass

```diff
@@ -1207,6 +1207,12 @@ toolchain_src_configure() {
 		is-flagq -mfloat-gprs=double && confgcc+=( --enable-e500-double )
 		[[ ${CTARGET//_/-} == *-e500v2-* ]] && confgcc+=( --enable-e500-double )
 		;;
+	ppc64)
+		# On ppc64 big endian target gcc assumes elfv1 by default,
+		# and elfv2 on little endian
+		# but musl does not support elfv1 at all here
+		[[ ${CTARGET//_/-} == *-musl ]] && confgcc+=( --with-abi=elfv2 )
```

Comment:
do you need `_-` mangling here? Looks like you don't. if `LE` does not need an override I suggest narrowing it work to a broken tuple: `[[ ${CTARGET} == powerpc64-*-musl ]]`. Can you link a gcc bug here as well? Default should just work upstream unless musl plans to implement elfv1.

I also don't understand what `here` means in `but musl does not support elfv1 at all here`. On powerpc64{,le}-?

Also note: clang does the same:
```
$ clang -target  powerpc64le-unknown-musl -dM -E -x c - </dev/null | fgrep ELF
#define _CALL_ELF 2
$ clang -target  powerpc64-unknown-musl -dM -E -x c - </dev/null | fgrep ELF
#define _CALL_ELF 1
```
and will need to adapt upstream as well.

Reply:
here means on ppc64 yeah, will reword.

there's no gcc bug to link, it's sort of common knowledge in musl circles.
and I don't think they going to implement elfv1.


regarding `le`, gcc will already configure itself to use v2, should we really avoid passing it on `le`? 
or explicitly pass it for consistency?

basically question is `powerpc64-*musl` vs `powerpc64*-musl`


as for clang, @q66  said it needs patching to do V2 on `powerpc64-*-musl`

sample patch here, not sure if it's stand-alone or requires other patches which add custom targets
https://github.com/void-ppc/void-packages/blob/master/srcpkgs/llvm9/files/patches/clang/clang-004-ppc64-musl-elfv2.patch

Reply:
> there's no gcc bug to link, it's sort of common knowledge in musl circles.
> and I don't think they going to implement elfv1.

I find it odd to have upstream toolchain broken by default and every downstream having to workaround it. Is there at least musl page that explains how to get a working toolchain?

> regarding `le`, gcc will already configure itself to use v2, should we really avoid passing it on `le`?
> or explicitly pass it for consistency?
> basically question is `powerpc64-*-musl` vs `powerpc64*-musl`

I'm leaning towards `powerpc64-*-musl` as it's a workaround and should go away. It' easier to reason about it it scope is smaller.

> as for clang, @q66 said it needs patching to do V2 on `powerpc64-*-musl`
> 
> sample patch here, not sure if it's stand-alone or requires other patches which add custom targets
> https://github.com/void-ppc/void-packages/blob/master/srcpkgs/llvm9/files/patches/clang/clang-004-ppc64-musl-elfv2.patch

Looks like it's only tuple recognition. Should be upstreamable in some form.

Reply:
Clang needs more patching for musl and ELFv2, https://github.com/void-ppc/void-packages/blob/master/srcpkgs/llvm9/files/patches/clang/clang-002-add-musl-triples.patch and https://github.com/void-ppc/void-packages/blob/master/srcpkgs/llvm9/files/patches/clang/clang-003-ppc64-dynamic-linker-path.patch is also needed. There are also patches needed within LLVM itself, you can take a look in the `llvm` directory.

Reply:
That said, our patches assume v2 unconditionally, as we also use it for glibc. If you want to keep v1 on glibc, you will need to do more elaborate patching.